### ISSUE TITLE
steamshovel times relative to particle (Trac #2061)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2053]
When I set the event times according to a particle (lower-left drop down menu), it seems that steamshovel now assumes that the time interval starts at a depth of about 1000 m. This is convenient for in-ice, but not so for IceTop since the interval starts 3 microseconds late. Is this hardcoded somewhere? Each particle has a vertex properly set, perhaps one can use that vertex somehow?

I'm using offline-software V17-05-00.

Migrated from https://code.icecube.wisc.edu/ticket/2061
```json
{
    "status": "closed", 
    "changetime": "2017-08-18T07:22:34", 
    "description": "When I set the event times according to a particle (lower-left drop down menu), it seems that steamshovel now assumes that the time interval starts at a depth of about 1000 m. This is convenient for in-ice, but not so for IceTop since the interval starts 3 microseconds late. Is this hardcoded somewhere? Each particle has a vertex properly set, perhaps one can use that vertex somehow?\n\nI'm using offline-software V17-05-00.", 
    "reporter": "jgonzalez", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1503040954763004", 
    "component": "combo core", 
    "summary": "steamshovel times relative to particle", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2017-08-01T15:52:29", 
    "milestone": "", 
    "owner": "sander.vanheule", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170801T16:01:38: jgonzalez commented:

> Another option is to take the interval between the time a co-moving plane first hits the detector until it exits. This matters, for example: for a very inclined air shower placed on the surface, no need to show times when the bundle is still far away, and IceTop pulses might still happen before in-ice pulses, so one would take a rectangle and identify the first and last corners to be hit by the plane.

**Comment by icecube-trac** [comment]
Trac update at 20170801T16:12:18: 
* olivas changed status from "new" to "assigned"
* olivas changed owner from "" to "sander.vanheule"
* olivas changed component from "other" to "icecube offline"

**Comment by icecube-trac** [comment]
Trac update at 20170818T07:22:34: 
* sander.vanheule changed resolution from "" to "fixed"
* sander.vanheule changed status from "assigned" to "closed"
* sander.vanheule commented:

> In changeset 157659:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="157659"
> Improve time range determination for tracks.
> 
> Instead of using a (±1km)² cubic volume around the detector centre,
> use a smaller cylindrical volume around IceCube, and represent IceTop
> as a disc. Depending on the direction of the track, automatic ranging
> will now try to include IceTop.
> 
> Passing times for IceCube are determined by intersecting the track
> with the cylinder. Passing times for IceTop are determined by determining
> the time instants where a plane perpendicular to the track touches
> the IceTop disc.
> 
> Fixes #2053.
> ```
